services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/ProjectSettingsManagerImpl.java

```diff
@@ -306,36 +306,42 @@ private void validateS3BucketAccess(ExternalS3StorageLocationSetting externalS3S
 
 	private void validateS3BucketOwnership(ExternalS3StorageLocationSetting externalS3StorageLocationSetting, List<PrincipalAlias> ownerAliases)
 			throws IOException, NotFoundException {
+		// PLFM-5769: Check that the key is valid
```

Comment:
This method should have had a unit test.  That is technical debt that we now must pay to fix this issue.